### PR TITLE
shift AddTransaction from UserControler  to TransactionControler, fix…

### PR DESCRIPTION
**Pull request by MariuszMalec**
transaction works well , fix balance user according button edit and delete click in transaction view

### CODE REVIEW

File: TheBTeam.BLL/Services/TransactionService.cs [hunk 1]
```diff
@@ -111,12 +53,26 @@ public static void ApplyTransaction(Transaction transaction, User user)
...
-            return Transaction.Where(t => t.User.Id == id).ToList();
+            return _transactions.Where(t => t.User.Id == id).ToList();
         }
         public static List<Transaction> SearchTransactionByType(TypeOfTransaction type, List<Transaction> transactions)
         {
             var transactionsByType = transactions.Where(t => t.Type == type).ToList();
             return transactionsByType;
         }
+
+        public Transaction GetTransactionByUser(string id)
```
Comment (PuchalskiMateusz) [inline TheBTeam.BLL/Services/TransactionService.cs:64]: so here is GetTransactionByUser and later on is just Delete, we should keep one pattern
According to the fact, that we are working here on non-static class I thing less is better like GetByUser

File: TheBTeam.Web/Controllers/TransactionController.cs [hunk 2]
```diff
@@ -13,22 +13,24 @@ namespace TheBTeam.Web.Controllers
 {
     public class TransactionController : Controller
     {
-        private TransactionService _transactionService;//TODO wez tutaj tranzakcje z usera
+        private TransactionService _transactionService;
+        private UserService _userService;
         public TransactionController()
         {
             _transactionService = new TransactionService();
+            _userService = new UserService();
```
Comment (PuchalskiMateusz) [inline TheBTeam.Web/Controllers/TransactionController.cs:21]: are you shure, that in transactionService should be userservice?

File: TheBTeam.Web/Controllers/TransactionController.cs [hunk 3]
```diff
@@ -13,22 +13,24 @@ namespace TheBTeam.Web.Controllers
...
-            //TODO how to get here users!
             var model = _transactionService.GetAll(category, type);
             return View(model);
         }
         // GET: TransactionController/Details/5
-        public ActionResult Details(int id)
+        public ActionResult Details(string id)
         {
-            return View();
+            var model = _transactionService.GetTransactionByEmail(id);
```
Comment (PuchalskiMateusz) [inline TheBTeam.Web/Controllers/TransactionController.cs:32]: getByEmail, but you pass Id

File: TheBTeam.Web/Controllers/TransactionController.cs [hunk 4]
```diff
@@ -45,6 +47,33 @@ public ActionResult UserTransactions()
...
+            try
+            {
+                if (!ModelState.IsValid)
+                {
+                    return View(model);
+                }
+                var id = TempData["id"] as string;
+                var user = _userService.GetById(id);
+                _transactionService.AddTransactionByUser(model, user);
+                return RedirectToAction(nameof(Index));
```
Comment (PuchalskiMateusz) [inline TheBTeam.Web/Controllers/TransactionController.cs:69]: i thing that there we should go to users transaction, not all of them

Comment (MariuszMalec) [reply TheBTeam.Web/Controllers/TransactionController.cs]: It's a little strange but this works well and go to user transaction

**Review by MichalMichanski — APPROVED** [review]
Looks good, pulled it and failed to break it, i think we should get rid of uneditable props from edit user but that goes beyond the topic of this pr, merge away if you ask me.

**Pull request merged**